Formatter/Pool.php

```diff
@@ -39,10 +39,19 @@ class Pool
      */
     public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
     {
+        // NEXT_MAJOR: make both parameters required
```

Comment:
Do we really need a logger here? Personally I would prefer leaving it optional and using the NullLogger as a fallback so that there is no need to check the existence of the logger. I would even consider removing it from the constructor and using the LoggerAwareInterface.